cpp/src/groupby/sort/group_rank_scan.cu

```diff
@@ -98,7 +99,9 @@ std::unique_ptr<column> rank_scan(column_view const& order_by,
     order_by,
     group_labels,
     group_offsets,
-    [] __device__(bool equality, auto row_index) { return equality ? row_index + 1 : 0; },
+    [] __device__(bool unequal, auto row_index_in_group) {
+      return unequal ? row_index_in_group + 1 : 0;
```

Comment:
Wait, the logic is inverted in the new code. Is this intentional?

Reply:
Only the name is inverted, to reflect the correct logic. You'll notice that the calling code passes `!comparator(...)`.

The old name `equality` implies `is_equal`, which is incorrect in this case. We need to increment the rank when the rows are unequal.